Tests/Logging/ReadabilityExtensionTests.cs

```diff
@@ -0,0 +1,103 @@
+
+            Assert.That(typeof(TestBigGenericClass<int, int, int, int, int, int, int, int, int, int>).GetShallowGenericTypeCount(), Is.EqualTo(10));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int>.TestInnerClass).GetShallowGenericTypeCount(), Is.EqualTo(0));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int>.TestInnerGenericClass<float>).GetShallowGenericTypeCount(), Is.EqualTo(1));
+        }
+
+        [Test]
+        public static void GetReadableNameTest()
+        {
+            Assert.That(typeof(int).GetReadableName(), Is.EqualTo("Int32"));
+
+            Assert.That(typeof(int?).GetReadableName(), Is.EqualTo("Int32?"));
+
+            Assert.That(typeof(TestClass).GetReadableName(), Is.EqualTo("ReadabilityExtensionTests.TestClass"));
+
+            Assert.That(typeof(TestGenericClass<int>).GetReadableName(), Is.EqualTo("ReadabilityExtensionTests.TestGenericClass<Int32>"));
+
+            Assert.That(typeof(TestGenericClass<int?>).GetReadableName(), Is.EqualTo("ReadabilityExtensionTests.TestGenericClass<Int32?>"));
+
+            Assert.That(typeof(TestGenericClass<TestGenericClass<TestClass>>).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestGenericClass<ReadabilityExtensionTests.TestGenericClass<ReadabilityExtensionTests.TestClass>>"));
+
+            Assert.That(typeof(TestBigGenericClass<int, int, int, int, int, int, int, int, int, int>).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestBigGenericClass<Int32, Int32, Int32, Int32, Int32, Int32, Int32, Int32, Int32, Int32>"));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int>.TestInnerClass).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestOuterGenericNestedClass<Int32>.TestInnerClass"));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int>.TestInnerGenericClass<float>).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestOuterGenericNestedClass<Int32>.TestInnerGenericClass<Single>"));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int?>.TestInnerClass).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestOuterGenericNestedClass<Int32?>.TestInnerClass"));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<int?>.TestInnerGenericClass<float?>).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestOuterGenericNestedClass<Int32?>.TestInnerGenericClass<Single?>"));
+
+            Assert.That(typeof(TestOuterGenericNestedClass<>.TestInnerGenericClass<>).GetReadableName(),
+                Is.EqualTo("ReadabilityExtensionTests.TestOuterGenericNestedClass<>.TestInnerGenericClass<>"));
+        }
+
+        [Test]
+        public static void GetDigitCount()
+        {
+            Assert.That(0.GetDigitCount(), Is.EqualTo(1));
+            Assert.That(0.GetDigitCount(true), Is.EqualTo(1));
```

Comment:
Neat, i didn't know you could call methods on a literal.

Reply:
Yea, you do have to be careful about expressions like this though.
`-1.GetDigitCount(true)`. The `-` gets applied after the method call.

The fix is
`(-1).GetDigitCount(true)`